openmls/src/test_utils/test_framework/messages.rs

```diff
@@ -0,0 +1,63 @@
+//! This module contains code to convert messages between the
+//! `MlsMessage`/`MlsMessageIn` formats as used by the Managed API, which the
+//! Clients are built on and the `DsMlsMessage`, which includes the message type
+//! in its serialized form.
+
+use crate::{
+    group::MlsMessageIn,
+    prelude::{MlsCiphertext, VerifiableMlsPlaintext},
+};
+
+use tls_codec::{TlsDeserialize, TlsSerialize, TlsSize};
+
+/// Enum defining encodings for the different message types/
+#[derive(Debug, Clone, Copy, TlsSerialize, TlsDeserialize, TlsSize)]
+#[repr(u8)]
+pub enum MessageType {
+    /// An MlsCiphertext message.
+    MlsCiphertext = 0,
+
+    /// An MlsPlaintext message.
+    MlsPlaintext = 1,
+}
+
+impl<'a> tls_codec::Size for MlsMessageIn<'a> {
```

Comment:
Unrelated but if you need something to do and want to poke around the tls codec code; This pattern of implementing (de)serialization of enums could be put into the library and would safe us quite some code.